sql/catalyst/src/main/scala/org/apache/spark/sql/Row.scala

```diff
@@ -325,6 +341,14 @@ trait Row extends Serializable {
   def getAs[T](i: Int): T = get(i).asInstanceOf[T]
 
   /**
+   * Returns the Optional value of a given fieldName.
+   * If the value is null or if it cannot be cast to the requested type, None is returned. None is
+   * also returned if either the schema is not defined or else the given fieldName does not exist
+   * in the schema.
+   */
+  def getAsOpt[T](fieldName: String): Option[T] = None
```

Comment:
Yeah, but this is data coming out of Spark SQL, so the type will always be correct.  Bad data would not make it this far.